l10n.js

```diff
@@ -219,40 +220,75 @@ document.webL10n = (function(window, document, undefined) {
-            if (reComment.test(line))
-              continue;
-
-            // the extended syntax supports [lang] sections and @import rules
-            if (extendedSyntax) {
-              match = reSection.exec(line);
-              if (match) { // section start?
-                // RFC 4646, section 4.4, "All comparisons MUST be performed
-                // in a case-insensitive manner."
-
-                currentLang = match[1].toLowerCase();
-                skipLang = (currentLang !== '*') &&
-                    (currentLang !== lang) && (currentLang !== genericLang);
-                continue;
-              } else if (skipLang) {
-                continue;
-              }
-              match = reImport.exec(line);
-              if (match) { // @import rule?
-                loadImport(baseURL + match[1], nextEntry);
-                return;
-              }
-            }
 
-            // key-value pair
+            var line = processLine(entries.shift(), extendedSyntax);
+
             var tmp = line.match(reSplit);
             if (tmp && tmp.length == 3) {
               dictionary[tmp[1]] = evalString(tmp[2]);
+
+              if (hasEOLDelimeter(line)) {
+                dictionary[tmp[1]] += evalString(processMultiLine(entries.shift(), extendedSyntax));
+              }
             }
           }
         }
+
+        // Process a line and check to see if it has an EOL delimiter. If it does, check the next line.
+        function processMultiLine(line, extendedSyntax) {
+          line = processLine(line, extendedSyntax);
+
+          // Does this line have an EOL delimeter?
+          if (hasEOLDelimeter(line)) {
+            // Append the next line to our current line and return
+            return line + '\n' + processMultiLine(entries.shift(), extendedSyntax);
+          } else { // There is no EOL delimeter
+            return line;
+          }
+        }
+
+        // Process a line and return a string
+        function processLine(line, extendedSyntax) {
+          line = line.replace(reBlank, '');
+
+          var match;
+
+          // comment or blank line?
+          if (reComment.test(line))
+            return '';
+
+          // the extended syntax supports [lang] sections and @import rules
+          if (extendedSyntax) {
+            match = reSection.exec(line);
+            if (match) { // section start?
+              // RFC 4646, section 4.4, "All comparisons MUST be performed
+              // in a case-insensitive manner."
+
+              currentLang = match[1].toLowerCase();
+              skipLang = (currentLang !== '*') &&
+                  (currentLang !== lang) && (currentLang !== genericLang);
+              return '';
+            } else if (skipLang) {
+              return '';
+            }
+            match = reImport.exec(line);
+            if (match) { // @import rule?
+              loadImport(baseURL + match[1], nextEntry);
+              return '';
```

Comment:
These two lines are problematic, because `loadImport` may be asynchronous (that's why the `nextEntry` callback is passed, and the original code uses `return` instead of `continue`).